iroh-sync/src/sync.rs

```diff
@@ -242,40 +239,28 @@ pub enum CapabilityError {
     NamespaceMismatch,
 }
 
-/// Local representation of a mutable, synchronizable key-value store.
+/// In memory information about an open replica.
 #[derive(derive_more::Debug)]
-pub struct Replica<S: ranger::Store<SignedEntry> + PublicKeyStore + store::DownloadPolicyStore> {
-    capability: Capability,
-    peer: Peer<SignedEntry, S>,
+pub struct ReplicaInfo {
+    pub(crate) capability: Capability,
     subscribers: Subscribers,
     #[debug("ContentStatusCallback")]
     content_status_cb: Option<ContentStatusCallback>,
     closed: bool,
 }
 
-impl<S: ranger::Store<SignedEntry> + PublicKeyStore + store::DownloadPolicyStore + 'static>
-    Replica<S>
-{
+impl ReplicaInfo {
     /// Create a new replica.
-    pub fn new(capability: Capability, store: S) -> Self {
-        Replica {
+    pub fn new(capability: Capability) -> Self {
+        Self {
             capability,
-            peer: Peer::from_store(store),
             subscribers: Default::default(),
             // on_insert_sender: RwLock::new(None),
             content_status_cb: None,
             closed: false,
         }
     }
 
-    /// Mark the replica as closed, prohibiting any further operations.
-    ///
-    /// This method is not public. Use [store::Store::close_replica] instead.
-    pub(crate) fn close(&mut self) {
```

Comment:
why is this rmeoved?

Reply:
This was called by close_replica, but is not needed anymore.

It is setting the closed flag to true and clears the subscribers, but it was called on an instance that was immediately dropped afterward, so it really did nothing as far as I can see. Maybe it did something at some point, but not anymore after some refactoring.

Reply:
Yup, this was important before we had the actor as replicas were cloned freely then. Now the actor owns the only instance so close equals drop.